pym/bob/cmds/build/build.py

```diff
@@ -156,6 +158,20 @@ def _downloadArgument(arg):
         elif args.jobs <= 0:
             parser.error("--jobs argument must be greater than zero!")
 
+        # parse MAKEFLAGS environment variable to setup number of jobs
+        # when called from make
+        makeFlags = os.environ.get('MAKEFLAGS')
+        makeFds = None
+        if makeFlags is not None:
+            jobs = re.search(r'-j([0-9]*)', makeFlags)
+            fds = re.search(r'--jobserver-auth=([0-9]*),([0-9]*)', makeFlags)
+            if jobs and fds and jobs.group(1) and fds.group(1) and fds.group(2):
+                if noJobs:
+                    args.jobs = int(jobs.group(1))
+                    makeFds = [int(fds.group(1)), int(fds.group(2))]
+                else:
+                    print("warning: -j" + str(args.jobs) + " forced: resetting jobserver mode.");
+
```

Comment:
I think we have to check if the file descriptors are actually open. If the `bob` call is not prefixed with `+` in the `Makefile` then `make` will still set `MAKEFLAGS` but the file descriptors are not open. Something along these lines should work (untested):

```python
try:
  if not all(stat.S_ISFIFO(os.stat(f).st_mode) for f in makeFds): makeFds = None
except OSError:
  makeFds = None
```